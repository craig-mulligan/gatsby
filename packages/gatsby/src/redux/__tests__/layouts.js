const reducer = require(`../reducers/pages`)
const { actions } = require(`../actions`)

describe(`Add layouts`, () => {
  it(`allows you to add layouts`, () => {
    const action = actions.createLayout(
      {
        id: `index`,
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add pages with context`, () => {
    const action = actions.createLayout(
      {
        id: `index`,
        component: `/whatever/index.js`,
        context: {
          title: `layout title`,
        },
      },
      { name: `test` }
    )
    const state = reducer(undefined, action)
    expect(action).toMatchSnapshot()
    expect(state).toMatchSnapshot()
  })

  it(`allows you to add multiple pages`, () => {
    const action = actions.createLayout(
      {
        id: `index`,
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    const action2 = actions.createLayout(
      {
        id: `indexpizza/`,
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(2)
  })

  it(`allows you to update existing pages (based on id)`, () => {
    const action = actions.createLayout(
      {
        id: `index`,
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )

    // Change the component
    const action2 = actions.createLayout(
      {
        id: `index`,
        component: `/whatever2/index.js`,
      },
      { name: `test` }
    )

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(1)
  })

  it(`allows you to delete ids`, () => {
    const action = actions.createLayout(
      {
        id: `index`,
        component: `/whatever/index.js`,
      },
      { name: `test` }
    )
    const action2 = actions.deletePage({ id: `index` })

    let state = reducer(undefined, action)
    state = reducer(state, action2)
    expect(state).toMatchSnapshot()
    expect(state.length).toEqual(0)
  })
})
