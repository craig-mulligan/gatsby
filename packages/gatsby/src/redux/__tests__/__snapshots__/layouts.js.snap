// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add layouts allows you to add layouts 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "context": Object {},
    "id": "index",
    "internalComponentName": "Component-layout-Index",
    "jsonName": "layout-index.json",
  },
  "plugin": Object {
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_LAYOUT",
}
`;

exports[`Add layouts allows you to add layouts 2`] = `Array []`;

exports[`Add layouts allows you to add multiple pages 1`] = `Array []`;

exports[`Add layouts allows you to add pages with context 1`] = `
Object {
  "payload": Object {
    "component": "/whatever/index.js",
    "componentChunkName": "component---whatever-index-js",
    "context": Object {
      "title": "layout title",
    },
    "id": "index",
    "internalComponentName": "Component-layout-Index",
    "jsonName": "layout-index.json",
  },
  "plugin": Object {
    "name": "test",
  },
  "traceId": undefined,
  "type": "CREATE_LAYOUT",
}
`;

exports[`Add layouts allows you to add pages with context 2`] = `Array []`;

exports[`Add layouts allows you to delete ids 1`] = `Array []`;

exports[`Add layouts allows you to update existing pages (based on id) 1`] = `Array []`;
